Keda.Cosmosdb.Scaler/src/Services/ExternalScalerService.cs

```diff
@@ -0,0 +1,202 @@
+﻿using System;
+using System.Data.Common;
+using System.Threading.Tasks;
+using Google.Protobuf.Collections;
+using Grpc.Core;
+using Keda.Cosmosdb.Scaler.Protos;
+using Microsoft.Azure.Documents.ChangeFeedProcessor;
+using Microsoft.Azure.Documents.Client;
+
+namespace Keda.Cosmosdb.Scaler.Services
+{
+    public class ExternalScalerService : ExternalScaler.ExternalScalerBase
+    {
+        private static ConsoleLogger logger = new ConsoleLogger();
+        public override async Task<IsActiveResponse> IsActive(ScaledObjectRef request, ServerCallContext context)
+        {
+            bool isActive = false;
+            try
+            {
+                var trigger = CreateTriggerFromMetadata(request.ScalerMetadata);
+                var workToBeDone = await GetEstimatedWork(trigger);
+
+                isActive = workToBeDone > 0;
+
+                if (isActive)
+                {
+                    logger.Log(string.Format("Activating to 1 for cosmosDB database {0} collection {1}", trigger.DatabaseName, trigger.CollectionName));
+                }
+            }
+            catch(Exception ex)
+            {
+                logger.Log(ex.Message);
+            }
+
+            return new IsActiveResponse
+            {
+                Result = isActive
+            };
+        }
+
+        public override Task<GetMetricSpecResponse> GetMetricSpec(ScaledObjectRef request, ServerCallContext context)
+        {
+            var resp = new GetMetricSpecResponse();
+            resp.MetricSpecs.Add(new MetricSpec
+            {
+                MetricName = "WorkToBeDone",
```

Comment:
The metric name needs to be unique across the cluster. so maybe something like `Normalize($"cosmosdb-{request.Name}-{metadata.DatabaseName}-{metadata.CollectionName}")`. This is the normalize method we use in KEDA itself. https://github.com/kedacore/keda/blob/main/pkg/util/normalize_string.go#L18-L24

Reply:
@ahmelsayed, I think we will still have same metric name if two separate change feeds are created to monitor the same Cosmos DB container. We should instead consider lease account/database/container and change-feed processor-name (called metric prefix here) to come up with unique names.